src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/TripPatternForDate.java

```diff
@@ -172,7 +172,13 @@ public TripPatternForDate newWithFilteredTripTimes(Predicate<TripTimes> filter)
       .filter(frequencyEntry -> filter.test(frequencyEntry.tripTimes))
       .collect(Collectors.toList());
 
-    if (filteredTripTimes.isEmpty() && !hasFrequencies()) {
+    final boolean hasFreqs = hasFrequencies();
+
+    if (filteredTripTimes.isEmpty() && !hasFreqs) {
+      return null;
+    }
+
+    if (hasFreqs && filteredTripTimes.isEmpty() && filteredFrequencies.isEmpty()) {
       return null;
     }
```

Comment:
Something like this maybe, but this seems a bit more complicated than it actually is. The issue is that this is very performance sensitive code, and we need to be quite careful what is calculated each time.

```suggestion
    if (filteredTripTimes.isEmpty()) {
      if (hasFrequencies()) {
        if (filteredFrequencies.isEmpty()) return null;
      } else {
        return null;
      }
    }
```